posh-git.psm1

```diff
@@ -11,26 +11,79 @@ if ($psv.Major -lt 3 -and !$NoVersionWarn) {
     "To suppress this warning, change your profile to include 'Import-Module posh-git -Args `$true'.")
 }
 
-Push-Location $psScriptRoot
-.\CheckVersion.ps1 > $null
+& $PSScriptRoot\CheckVersion.ps1 > $null
 
-. .\Utils.ps1
-. .\GitUtils.ps1
-. .\GitPrompt.ps1
-. .\GitTabExpansion.ps1
-. .\TortoiseGit.ps1
-Pop-Location
+. $PSScriptRoot\Utils.ps1
+. $PSScriptRoot\GitUtils.ps1
+. $PSScriptRoot\GitPrompt.ps1
+. $PSScriptRoot\GitTabExpansion.ps1
+. $PSScriptRoot\TortoiseGit.ps1
 
 if (!$Env:HOME) { $Env:HOME = "$Env:HOMEDRIVE$Env:HOMEPATH" }
 if (!$Env:HOME) { $Env:HOME = "$Env:USERPROFILE" }
 
 Get-TempEnv 'SSH_AGENT_PID'
 Get-TempEnv 'SSH_AUTH_SOCK'
 
-Export-ModuleMember `
-    -Alias @(
-        '??') `
-    -Function @(
+# Get the default prompt definition.
+if ($psv.Major -eq 2) {
+    $defaultPromptDef = "`$(if (test-path variable:/PSDebugContext) { '[DBG]: ' } else { '' }) + 'PS ' + `$(Get-Location) + `$(if (`$nestedpromptlevel -ge 1) { '>>' }) + '> '"
+}
+else {
+    $defaultPromptDef = [Runspace]::DefaultRunspace.InitialSessionState.Commands['prompt'].Definition
+}
+
+# If there is no prompt function or the prompt function is the default, export the posh-git prompt function.
+$promptReplaced = $false
+$currentPromptDef = if ($funcInfo = Get-Command prompt -ErrorAction SilentlyContinue) { $funcInfo.Definition }
+if (!$currentPromptDef -or ($currentPromptDef -eq $defaultPromptDef)) {
+    Set-Item Function:\prompt -Value {
+        $origLastExitCode = $LASTEXITCODE
+
+        # A UNC path has no drive so it's better to use the ProviderPath e.g. "\\server\share".
+        # However for any path with a drive defined, it's better to use the Path property.
+        # In this case, ProviderPath is "\LocalMachine\My"" whereas Path is "Cert:\LocalMachine\My".
+        # The latter is more desirable.
+        $pathInfo = $ExecutionContext.SessionState.Path.CurrentLocation
+        $curPath = if ($pathInfo.Drive) { $pathInfo.Path } else { $pathInfo.ProviderPath }
+        if ($curPath -and $curPath.ToLower().StartsWith($Home.ToLower()))
+        {
+            $curPath = "~" + $curPath.SubString($Home.Length)
+        }
+
+        # Write the current path.
+        Write-Host $curPath -NoNewline
+
+        # Write the Git status summary information.
+        Write-VcsStatus
+
+        # When in debug mode, let user know
+        if ((Test-Path Variable:\PSDebugContext) -or [runspace]::DefaultRunspace.Debugger.InBreakpoint) {
+            $promptSuffix = "`n[DBG]: PS>> "
+        }
+        else {
+            $promptSuffix = "`nPS> "
+        }
+
+        $global:LASTEXITCODE = $origLastExitCode
+        $promptSuffix
+    }
+
+    $promptReplaced = $true

```

Comment:
Could we attach a property to our `prompt` instead of using a separate flag? I don't suppose there's any way to set its `Source = "posh-git"` like all the other functions we export?